sktime/classification/base.py

```diff
@@ -266,3 +282,295 @@ def _predict_proba(self, X) -> np.ndarray:
+            raise ValueError(
+                f"Mismatch in number of cases. Number in X = {n_cases} nos in y = "
+                f"{n_labels}"
+            )
+        if isinstance(y, np.ndarray):
+            if y.ndim > 1:
+                raise ValueError(
+                    f"y must be 1-dimensional but is in fact " f"{y.ndim} dimensional"
+                )
+
+
+def _get_data_characteristics(X):
+    """Query the data to find its characteristics for classifier capability check.
+
+    This is for checking array input where we assume series are equal length.
+    classifiers can take either ndarray or pandas input, and the checks are different
+    for each. For ndarrays, it checks:
+        a) whether x contains missing values;
+        b) whether x is multivariate.
+    for pandas it checks
+        a) whether x contains unequal length series;
+        a) whether x contains missing values;
+        b) whether x is multivariate.
+
+    Parameters
+    ----------
+    X : pd.pandas containing pd.Series or np.ndarray of either 2 or 3 dimensions.
+
+    Returns
+    -------
+    three boolean data characteristics: missing, multivariate and unequal
+    """
+    if isinstance(X, np.ndarray):
+        missing = _has_nans(X)
+        if X.ndim == 3 and X.shape[1] > 1:
+            multivariate = True
+        else:
+            multivariate = False
+        return missing, multivariate, False
+    else:
+        missing = _nested_dataframe_has_nans(X)
+        cols = len(X.columns)
+        if cols > 1:
+            multivariate = True
+        else:
+            multivariate = False
+        unequal = _nested_dataframe_has_unequal(X)
+        return missing, multivariate, unequal
+
+
+def _nested_dataframe_has_unequal(X: pd.DataFrame) -> bool:
+    """Check whether an input nested DataFrame of Series has unequal length series.
+
+    Parameters
+    ----------
+    X : pd.DataFrame where each cell is a pd.Series
+
+    Returns
+    -------
+    True if x contains any NaNs, False otherwise.
+    """
+    rows = len(X)
+    cols = len(X.columns)
+    s = X.iloc[0, 0]
+    length = len(s)
+
+    for i in range(0, rows):
+        for j in range(0, cols):
+            s = X.iloc[i, j]
+            temp = len(s)
+            if temp != length:
+                return True
+    return False
+
+
+def _nested_dataframe_has_nans(X: pd.DataFrame) -> bool:
+    """Check whether an input pandas of Series has nans.
+
+    Parameters
+    ----------
+    X : pd.DataFrame where each cell is a pd.Series
+
+    Returns
+    -------
+    True if x contains any NaNs, False otherwise.
+    """
+    cases = len(X)
+    dimensions = len(X.columns)
+    for i in range(0, cases):
+        for j in range(0, dimensions):
+            s = X.iloc[i, j]
+            for k in range(0, s.size):
+                if pd.isna(s[k]):
+                    return True
+    return False
+
+
+# @njit(cache=True)
+def _has_nans(x: np.ndarray) -> bool:
```

Comment:
This whole function can be done using .any

```
def _has_nans(x: np.ndarray) -> bool
    return np.isnan(x).any()
```

Reply:
I'm sure there was a reason I didnt want to do that, possibly for the early abandon? I imagine franz will replace this anyway, so I will leave it for now but not commit before sorting this out (I'll run some tests if necessary)

Reply:
so I remember now, the problem with np.isnan is it returns an array of booleans, which a) is wasteful in memory and b) does not allow the early abandon with the first nan. But I am not really fussed either way, its unlikely to be a huge overhead

Reply:
I've replaced this by faster functionality in the PR I made, have you looked at that?